src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs

```diff
@@ -227,35 +230,39 @@ public override ImmutableArray<CustomModifier> RefCustomModifiers
 
         private sealed class SynthesizedComplexParameterSymbol : SynthesizedParameterSymbolBase
         {
-            private readonly ImmutableArray<CustomModifier> _refCustomModifiers;
-            private readonly ImmutableArray<CSharpAttributeData> _attributes;
+            private readonly ParameterSymbol _baseParameter;
+            private readonly bool _inheritAttributes;
+
 
             public SynthesizedComplexParameterSymbol(
                 MethodSymbol container,
                 TypeWithAnnotations type,
-                int ordinal,
-                RefKind refKind,
-                string name,
-                ImmutableArray<CustomModifier> refCustomModifiers,
-                ImmutableArray<CSharpAttributeData> attributes)
-                : base(container, type, ordinal, refKind, name)
+                ParameterSymbol baseParameter,
+                bool inheritAttributes)
+                : base(container,
+                      type,
+                      baseParameter.Ordinal,
+                      baseParameter.RefKind,
+                      baseParameter.Name)
             {
-                Debug.Assert(!refCustomModifiers.IsDefault);
-                Debug.Assert(!attributes.IsDefault);
+                _baseParameter = baseParameter;
+                _inheritAttributes = inheritAttributes;
 
-                _refCustomModifiers = refCustomModifiers;
-                _attributes = attributes;
             }
 
             public override ImmutableArray<CustomModifier> RefCustomModifiers
             {
-                get { return _refCustomModifiers; }
+                get { return _baseParameter.RefCustomModifiers; }
             }
 
             public override ImmutableArray<CSharpAttributeData> GetAttributes()
             {
-                return _attributes;
+                return _inheritAttributes ? _baseParameter.GetAttributes() : ImmutableArray<CSharpAttributeData>.Empty;
             }
+
+            internal override bool HasEnumeratorCancellationAttribute => _inheritAttributes ? _baseParameter.HasEnumeratorCancellationAttribute : false;
+
+            // PROTOTYPE: add overrides for other well-known parameter attributes used after lowering
```

Comment:
>PROTOTYPE [](start = 15, length = 9)

What feature this comment is tied to?

Reply:
MarshalAsAttribute IIRC, possibly others

Reply:
I'll add the local-function-attributes bit to it, thanks.